### PR TITLE
Add MainActivityTest and some changes and refactoring

### PR DESCRIPTION
**Pull request by alejandrocq**
Issue #28 

I have made this improvements:

* Update SDK and build tools versions.
* Update to Java 8 in order to use lambdas.
* Add MainActivityTest (interaction test) and the necessary libraries needed for testing.
* Changes and general refactoring in some parts of the code.

I have commited my .idea files too, but I think that you should remove that directory from git (usually is not a good idea to track IDE files, and also gradle files have almost all the information to get the developing environment ready).

I hope this is useful :).

### CODE REVIEW

**Comment by ashik94vc** [comment]
Hey, thanks a lot. Would you be able to add the idea files to gitignore too. I'll review your code tomorrow around morning and will merge it.

**Comment by alejandrocq** [comment]
> Hey, thanks a lot. Would you be able to add the idea files to gitignore too. I'll review your code tomorrow around morning and will merge it.

You're welcome! :) Yes, I can do that this afternoon. I can do another PR if you prefer to merge this before.

**Comment by alejandrocq** [comment]
I have added two commits before the changes to remove .idea folder files from git, and also I have simplified the .gitignore file a bit ;).

File: app/src/androidTest/java/com/cepheuen/elegantnumberbuttonsample/MainActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public void test_add_button_click() {
+        try {
+            // Change btn1 text to find it with ViewMatchers
+            rule.runOnUiThread(() -> activity.btn1.addBtn.setText("target"));
+        } catch (Throwable throwable) {
+            throwable.printStackTrace();
+        }
+        Espresso.onView(allOf(withId(R.id.add_btn), withText("target"))).perform(click());
+        assertThat(activity.btn1.getNumber(), Matchers.is("1"));
+    }
```
Comment (ashik94vc) [inline app/src/androidTest/java/com/cepheuen/elegantnumberbuttonsample/MainActivityTest.java]: We can add tests to cover other functions as well but this should surely be a good place to start.

File: app/src/main/java/com/cepheuen/elegantnumberbuttonsample/MainActivity.java [hunk 2]
```diff
@@ -1,96 +1,44 @@
 package com.cepheuen.elegantnumberbuttonsample;
 
 import android.graphics.Color;
-import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 import android.util.Log;
-import android.view.View;
 import android.widget.TextView;
-
+import androidx.appcompat.app.AppCompatActivity;
```
Comment (ashik94vc) [inline app/src/main/java/com/cepheuen/elegantnumberbuttonsample/MainActivity.java]: Good job in grouping the imports.

File: app/src/main/java/com/cepheuen/elegantnumberbuttonsample/MainActivity.java [hunk 3]
```diff
@@ -1,96 +1,44 @@
...
 import com.cepheuen.elegantnumberbutton.view.ElegantNumberButton;
```
Comment (ashik94vc) [inline app/src/main/java/com/cepheuen/elegantnumberbuttonsample/MainActivity.java]: Minor, adding a line break from previous imports is a good practice to isolate dependencies based on framework. It has zero impact on this code base though might be really helpful when the code base is huge.

**Pull request merged**